rosbag2_storage_default_plugins/test/rosbag2_storage_default_plugins/sqlite/storage_test_fixture.hpp

```diff
@@ -156,22 +161,45 @@ class StorageTestFixture : public TemporaryDirectoryFixture
   }
 
 protected:
-  int get_buffer_capacity(const std::string & message)
+  std::string get_preamble() const
+  {
+    return {0x00, 0x01, 0x00, 0x00, 0x0f, 0x00, 0x00, 0x00};
+  }
+
+  size_t get_buffer_capacity(const std::string & message)
   {
-    return write_data_to_serialized_string_message(nullptr, 0, message);
+    return get_preamble().size() + message.size();
   }
 
-  int write_data_to_serialized_string_message(
+  size_t write_data_to_serialized_string_message(
     uint8_t * buffer, size_t buffer_capacity, const std::string & message)
   {
-    // This function also writes the final null charachter, which is absent in the CDR format.
-    // Here this behaviour is ok, because we only test test writing and reading from/to sqlite.
-    return rcutils_snprintf(
-      reinterpret_cast<char *>(buffer),
-      buffer_capacity,
-      "%c%c%c%c%c%c%c%c%s",
-      0x00, 0x01, 0x00, 0x00, 0x0f, 0x00, 0x00, 0x00,
-      message.c_str());
+    size_t remaining_buffer = buffer_capacity;
+    size_t amount_written = 0;
+    if (remaining_buffer == 0) {
+      return amount_written;
+    }
+
+    size_t amount_to_write = remaining_buffer;
+    std::string preamble = get_preamble();
+    if (preamble.size() < amount_to_write) {
```

Comment:
@wjwwood nit: would it make sense to `assert(buffer_capacity >= preamble.size() + message.size())`?